packages/graphql/src/services/sessions.ts

```diff
@@ -50,6 +183,116 @@ export async function startSession(
   })
 }
 
+interface ActivateSessionBlockArgs {
+  sessionId: string
+  sessionBlockId: number
+}
+
+export async function activateSessionBlock(
+  { sessionId, sessionBlockId }: ActivateSessionBlockArgs,
+  ctx: ContextWithUser
+) {
+  const session = await ctx.prisma.session.findUnique({
+    where: {
+      id: sessionId,
+    },
+    include: { blocks: true },
+  })
+
+  if (!session || session.ownerId !== ctx.user.sub) return null
+
+  const newBlockIndex = session.blocks.findIndex(
+    (block) => block.id === sessionBlockId
+  )
+
+  // if the block is not from the current session, return early
+  if (newBlockIndex === -1) return session
+
+  // if the block is already active, return early
+  if (session.activeBlock === newBlockIndex) return session
+
+  // set the new block to active
+  const updatedSession = await ctx.prisma.session.update({
+    where: {
+      id: sessionId,
+    },
+    data: {
+      activeBlock: newBlockIndex,
+      blocks: {
+        updateMany: [
+          session.activeBlock > -1
+            ? {
+                where: {
+                  id: session.blocks[session.activeBlock].id,
+                },
+                data: {
+                  status: SessionBlockStatus.EXECUTED,
+                },
+              }
+            : undefined,
+          {
+            where: {
+              id: sessionBlockId,
+            },
+            data: {
+              status: SessionBlockStatus.ACTIVE,
+            },
+          },
+        ].filter(Boolean) as any[],
+      },
+    },
+    include: { blocks: true },
+  })
+
+  // get the details and instances of the newly activated session block
+  const newActiveBlock = await ctx.prisma.sessionBlock.findUnique({
+    where: {
+      id: sessionBlockId,
+    },
+    include: {
+      instances: true,
+    },
+  })
+
+  // initialize the cache for the new active block
+  const redisMulti = ctx.redisExec.multi()
```

Comment:
`multi()` together with `exec()` later on are starting a redis transaction. so all of the commands between `multi` and `exec` will be done in one request, and together.